lib/generators/base_generator.rb

```diff
@@ -0,0 +1,59 @@
+require "active_support/inflector"
+
+# Base class for all generators to come. If you want to override some behaviour, all you
+# got to do is override a method in the child class.
+class BaseGenerator < Rails::Generators::NamedBase
+  class_option :rspec, type: :boolean, desc: 'DEFAULT: Define rspec as your test framework.'
+  class_option :minitest, type: :boolean, desc: 'Test framework to generate test. (rspec or minitest)'
+  class_option :no_suffix, type: :boolean, desc: "If you don't want any pattern suffix. Ex: AuthenticationService. Service is the suffix."
+
+  # Creates the file for the requested pattern based on a template file.
+  def copy_service_file
```

Comment:
Is this used anywhere? Feels a bit odd that a method for a 'service' is implemented in a generic `BaseGenerator`.